From sshtylyov@ru.mvista.com  Fri Jul 10 10:33:25 2009
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Fri, 10 Jul 2009 20:02:44 +0300
Subject: USB: musb_host: undo incorrect change in musb_advance_schedule()
To: gregkh@suse.de
Cc: linux-usb@vger.kernel.org, david-b@pacbell.net, felipe.balbi@nokia.com
Message-ID: <200907102102.44733.sshtylyov@ru.mvista.com>
Content-Disposition: inline


Commit c9cd06b3d6ea825c62e277def929cc4315802b48 (musb_host: refactor
URB giveback) included due to my overlook the change incorrect in the
context of the current kernel -- undo it.

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/musb_host.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/musb/musb_host.c
+++ b/drivers/usb/musb/musb_host.c
@@ -373,7 +373,7 @@ static void musb_advance_schedule(struct
 		musb_save_toggle(qh, is_in, urb);
 		break;
 	case USB_ENDPOINT_XFER_ISOC:
-		if (urb->error_count)
+		if (status == 0 && urb->error_count)
 			status = -EXDEV;
 		break;
 	}
